library/spdm_common_lib/libspdm_com_opaque_data.c

```diff
@@ -220,15 +225,25 @@ spdm_process_opaque_data_supported_version_data(IN spdm_context_t *spdm_context,
          SECURED_MESSAGE_OPAQUE_ELEMENT_SMDATA_DATA_VERSION) ||
         (opaque_element_support_version->sm_data_id !=
          SECURED_MESSAGE_OPAQUE_ELEMENT_SMDATA_ID_SUPPORTED_VERSION) ||
-        (opaque_element_support_version->version_count != 1)) {
+        (opaque_element_support_version->version_count == 0)) {
         return RETURN_UNSUPPORTED;
     }
     versions_list = (void *)(opaque_element_support_version + 1);
-    if ((versions_list[0] >> SPDM_VERSION_NUMBER_SHIFT_BIT) != SPDM_MESSAGE_VERSION_11) {
+
+    result = spdm_negotiate_connection_version(&common_version, spdm_context->local_context.secured_message_version.spdm_version,
+                                    spdm_context->local_context.secured_message_version.spdm_version_count,
+                                    versions_list,
+                                    opaque_element_support_version->version_count);
+    if (result == FALSE) {
         return RETURN_UNSUPPORTED;
+    } else {
```

Comment:
Can remove the `else` statement since the `return RETURN_UNSUPPORTED;` at line 246 is not needed.